packages/@ember/-internals/runtime/lib/system/core_object.d.ts

```diff
@@ -1,15 +1,39 @@
-import { EmberClassConstructor, EmberInstanceArguments, Objectify } from './-private/types';
+/**
```

Comment:
This is also in the bucket of things I *think* we probably want to *not* add to core? 🤔 Though I do note you have it all as `@internal` and maybe it's useful to have internally but not publish. The trick will be that we probably also need to publish it in that secondary package, but having both may be a necessary evil for the transition period. cc. @jamescdavis @dfreeman for thoughts.

Reply:
This is in the internals package still and I do think we'll want it here. Whether we want to re-export it in @ember/object/core is another question.